core/drivers/stm32_uart.c

```diff
@@ -113,19 +114,32 @@ static void register_secure_uart(struct stm32_uart_pdata *pd)
 	size_t __maybe_unused n = 0;
 
 	stm32mp_register_secure_periph_iomem(pd->base.pa);
+#ifndef CFG_STM32MP25
```

Comment:
for consistency, could you condition also the above line with `#ifndef CFG_STM32MP25`
(and by the way, remove this useless leftover `n` local variable, if you don't mind)